drivers/sensor/pzem004t/pzem004t.c

```diff
@@ -0,0 +1,340 @@
+/*
+ * Copyright (c) 2025 Srishtik Bhandarkar
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT peacefair_pzem004t
+
+/*
+ * sensor pzem004t.c - Driver for peacefair PZEM004T sensor
+ * PZEM004T product: https://en.peacefair.cn/product/772.html
+ */
+
+#include <errno.h>
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/modbus/modbus.h>
+#include <zephyr/drivers/sensor/pzem004t.h>
+#include "pzem004t.h"
+
+LOG_MODULE_REGISTER(PZEM004T, CONFIG_SENSOR_LOG_LEVEL);
```

Comment:
Log module names are lowercase

```suggestion
LOG_MODULE_REGISTER(pzem004t, CONFIG_SENSOR_LOG_LEVEL);
```